src/SimulationRunner.cc

```diff
@@ -705,6 +705,15 @@ void SimulationRunner::Step(const UpdateInfo &_info)
   // Update all the systems.
   this->UpdateSystems();
 
+  if (!this->Paused() &&
+       this->requestedRunToSimTime >
+       std::chrono::steady_clock::duration::zero() &&
+       this->currentInfo.simTime >= this->requestedRunToSimTime)
+  {
+    this->SetPaused(true);
+    this->requestedSeek = std::chrono::steady_clock::duration{-1};
```

Comment:
Did you mean to reset `requestedRunToSimTime` here?

```suggestion
    this->requestedRunToSimTime = std::chrono::steady_clock::duration{-1};
```

Reply:
Fixed in 446515f03e930a97a7fb05b8764305296b3b5889